frontend/src/pages/Dashboard/StudentDashboard.tsx

```diff
@@ -0,0 +1,519 @@
+  color: ${INK};
+  font-family: "Rubik";
+  font-size: 18px;
+  font-weight: 700;
+  text-transform: lowercase;
+
+  :hover {
+    cursor: pointer;
+    text-decoration: underline;
+    opacity: 0.8;
+  }
+`;
+
+const getTitleOfButton = (user: User): string => {
+  switch (user.sessionId) {
+    case -1:
+      return "Begin pre-assessment";
+    case 9:
+      return "Congratulations on finishing the study";
+    default:
+      return (
+        (user.onAssessment ? "Continue session " : "Begin session ") +
+        (user.sessionId + 1)
+      );
+  }
+};
+
+const filterByThisWeek = (dayString: string) => {
+  let today = new Date(
+    new Date().getFullYear(),
+    new Date().getMonth(),
+    new Date().getDate()
+  );
+  let todayDayOfWeek = today.getDay();
+  let day = new Date(dayString);
+  let daysApart =
+    (today.getTime() - day.getTime()) / (1000 * 60 * 60 * 24);
+  let dayOfWeek = day.getDay();
+  return !(
+    today !== day &&
+    (dayOfWeek > todayDayOfWeek || daysApart >= 7)
+  );
+};
+
+const isDayActive = (
+  dayNumber: number,
+  daysActive: string[]
+): boolean => {
+  for (let index in daysActive) {
+    let dayString = daysActive[index];
+    let day = new Date(dayString);
+    let dayOfWeek = day.getDay();
+    if (dayOfWeek === dayNumber) {
+      return true;
+    }
+  }
+  return false;
+};
+
+const isToday = (dayNumber: number): boolean => {
+  let today = new Date();
+  let dayOfWeek = today.getDay();
+  return dayNumber === dayOfWeek;
+};
+
+interface StatParams {
+  number: number;
+  description: string;
+}
+
+interface DayParams {
+  name: string;
+  day: number;
+  daysActiveThisWeek: string[];
+}
+
+interface DayLabelType {
+  isToday?: boolean;
+}
+const DayOfWeek: FunctionComponent<DayParams> = ({
+  name,
+  day,
+  daysActiveThisWeek,
+}) => {
+  let istoday = isToday(day);
+  let isActive = isDayActive(day, daysActiveThisWeek);
+  return (
+    <DayContainer>
+      {isActive ? (
+        <Star src={star}></Star>
+      ) : (
+        <DotContainer>
+          <Dot src={ellipse}></Dot>
+        </DotContainer>
+      )}
+      <DayLabel isToday={istoday}>{name}</DayLabel>
+    </DayContainer>
+  );
+};
+
+const Stat: FunctionComponent<StatParams> = ({
+  number,
+  description,
+}) => {
+  return (
+    <ProgressBox>
+      <ProgressStatNumber>{number}</ProgressStatNumber>
+      <ProgressStatDescription>{description}</ProgressStatDescription>
+    </ProgressBox>
+  );
+};
+
+interface SessionCardParams {
+  sessionNumber: number;
+  image: string;
+  isComplete: boolean;
+}
+
+const SessionCard: FunctionComponent<SessionCardParams> = ({
+  sessionNumber,
+  image,
+  isComplete,
+}) => {
+  return (
+    <SessionBox isComplete={isComplete}>
+      <SessionImage src={image} />
+      <SessionNumber>session {sessionNumber}</SessionNumber>
+    </SessionBox>
+  );
+};
+
+const StudentDashboard: FunctionComponent<StudentDashboardParams> = ({
+  student,
+  showNextSession,
```

Comment:
It may be more descriptive if we change `showNextSession` to something like `isStudentView` or something. It seems like it is differentiating between the student view and the researcher view of the StudentDashboard. It would be helpful if we needed to add another researcher view specific feature